shell/browser/api/electron_api_menu_mac.mm

```diff
@@ -155,8 +166,26 @@
   // be done manually.
   base::mac::ScopedSendingEvent sendingEventScoper;
 
-  // Don't emit unresponsive event when showing menu.
-  [menu popUpMenuPositioningItem:item atLocation:position inView:view];
+  if (frame) {
+    auto* rwhvm = static_cast<content::RenderWidgetHostViewMac*>(
+        frame.value()
+            ->render_frame_host()
+            ->GetOutermostMainFrameOrEmbedder()
+            ->GetView());
+    RenderWidgetHostViewCocoa* cocoa_view = rwhvm->GetInProcessNSView();
+    view = cocoa_view;
+  }
+
+  NSEvent* dummy_event = [NSEvent mouseEventWithType:NSEventTypeRightMouseDown
```

Comment:
Can this be replaced with call to `ui::EventForPositioningContextMenuRelativeToWindow` ?

Reply:
Unfortunately not, my fun fact is I figured all this out and _then_ realized Chrome had a similar method. Unfortunately that method defaults to using `[NSApp currentEvent]` if it has the correct type, but we almost always need to make our own to ensure the coords match what the user provided, not the coords that the event happened to have last.

Reply:
> Unfortunately that method defaults to using [NSApp currentEvent] if it has the correct type

Ah missed that, thanks for clarifying!